cli/cmd/uninstall/uninstall_test.go

```diff
@@ -0,0 +1,187 @@
+	require.NoError(t, err)
+	require.Len(t, secrets.Items, 0)
+}
+
+func TestDeleteServiceAccounts(t *testing.T) {
+	c := getInitializedCommand(t)
+	c.kubernetes = fake.NewSimpleClientset()
+	sa := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "consul-test-sa1",
+			Labels: map[string]string{
+				"release": "consul",
+			},
+		},
+	}
+	sa2 := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "consul-test-sa2",
+			Labels: map[string]string{
+				"release": "consul",
+			},
+		},
+	}
+	_, err := c.kubernetes.CoreV1().ServiceAccounts("default").Create(context.TODO(), sa, metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = c.kubernetes.CoreV1().ServiceAccounts("default").Create(context.TODO(), sa2, metav1.CreateOptions{})
+	require.NoError(t, err)
+	err = c.deleteServiceAccounts("consul", "default")
+	require.NoError(t, err)
+	sas, err := c.kubernetes.CoreV1().ServiceAccounts("default").List(context.TODO(), metav1.ListOptions{})
+	require.NoError(t, err)
+	require.Len(t, sas.Items, 0)
+}
+
+func TestDeleteRoles(t *testing.T) {
+	c := getInitializedCommand(t)
+	c.kubernetes = fake.NewSimpleClientset()
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "consul-test-role1",
+			Labels: map[string]string{
+				"release": "consul",
+			},
+		},
+	}
+	role2 := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "consul-test-role2",
+			Labels: map[string]string{
+				"release": "consul",
+			},
+		},
+	}
+	_, err := c.kubernetes.RbacV1().Roles("default").Create(context.TODO(), role, metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = c.kubernetes.RbacV1().Roles("default").Create(context.TODO(), role2, metav1.CreateOptions{})
+	require.NoError(t, err)
+	err = c.deleteRoles("consul", "default")
+	require.NoError(t, err)
+	roles, err := c.kubernetes.RbacV1().Roles("default").List(context.TODO(), metav1.ListOptions{})
+	require.NoError(t, err)
+	require.Len(t, roles.Items, 0)
+}
+
+func TestDeleteRoleBindings(t *testing.T) {
+	c := getInitializedCommand(t)
+	c.kubernetes = fake.NewSimpleClientset()
+	rolebinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "consul-test-role1",
+			Labels: map[string]string{
+				"release": "consul",
+			},
+		},
+	}
+	rolebinding2 := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "consul-test-role2",
+			Labels: map[string]string{
+				"release": "consul",
+			},
+		},
+	}
+	_, err := c.kubernetes.RbacV1().RoleBindings("default").Create(context.TODO(), rolebinding, metav1.CreateOptions{})
+	require.NoError(t, err)
+	_, err = c.kubernetes.RbacV1().RoleBindings("default").Create(context.TODO(), rolebinding2, metav1.CreateOptions{})
+	require.NoError(t, err)
+	err = c.deleteRoleBindings("consul", "default")
+	require.NoError(t, err)
+	rolebindings, err := c.kubernetes.RbacV1().RoleBindings("default").List(context.TODO(), metav1.ListOptions{})
+	require.NoError(t, err)
+	require.Len(t, rolebindings.Items, 0)
+}
```

Comment:
I think we can also add tests for the `findExistingInstallation` function.

Reply:
This one is kind of tricky for similar reasons to https://github.com/hashicorp/consul-k8s/blob/e77da79f99bbd1038f7b3dc203c6492127341e6b/cli/cmd/install/install.go#L338. 

I think I can open a separate PR this week to refactor the helm logic out of `uninstall.findExistingInstallation` and `install.checkForPreviousInstallations`, and unit test the part that's unrelated to helm logic. The reason to pull it out into a separate PR is since I'll be focusing on testing within the acceptance tests PR we can add this additional testing there after a quick refactor.